### ISSUE TITLE
Secondary Spark table. Switched - Input based not working

### ISSUE THREAD

**Issue by 17Blue** [speeduino/speeduino#1270]
The secondary spark Table switching does not work using input based. 
![Spark Table2](https://github.com/user-attachments/assets/104fe840-7713-4573-82e5-62fdd97fc1e0)
Using the same pin i can get the secondary Fuel table working using input based (had to use external pull up resistor - the internal pull up resistor does not work).
I noticed that the I_O summary on Tuner Studio does not list the Spark2 input pin.
 
![I_O Summary](https://github.com/user-attachments/assets/a13a9eee-2e7e-4e1f-8427-93307661f167)
I am using the latest firmware Feb 2024 release with February 2024 Fix Pack 2.

**Comment by noisymime** [comment]
Can you pop up a log and your tune? I just gave it a quick try and it switched over OK on the bench.

I'll run  up your tune though and see if I can get the same behaviour. 

Will also make sure to add the pin to the IO summary as well

**Comment by 17Blue** [comment]
[CurrentTune.zip](https://github.com/user-attachments/files/18037204/CurrentTune.zip)
Attached is the tune.  Will upload the log once I have logged it.
Thanks.

**Comment by noisymime** [comment]
I just tried your tune and it switches over fine. I did not that the Secondary Spark table mode was set to off in the tune you posted, but I simply changed that to `Switched - Input based`, set the polarity to `LOW` and enabled the internal pullup. Then applying ground to pin 30, the ignition switched between the 2 tables and the `spark2Active` flag is being set. 

I'll close this for now as it appears to be a tune and/or hardware issue.
